### PR TITLE
Solution 1

### PR DESCRIPTION
**Pull request by likhithasai**
Took a TDD approach where I wrote smaple tests that failed and wrote code that then made the tests pass.

### CODE REVIEW

File: likhithasai/assignment-1/Anagram.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import java.util.Arrays;
+
+public class Anagram {
+
+    public boolean isAnagram(String str1, String str2) {
+		int n1 = str1.length();
```
Comment (schahar) [inline likhithasai/assignment-1/Anagram.java]: Nit : remove extra spaces.

File: likhithasai/assignment-1/Anagram.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import java.util.Arrays;
+
```
Comment (schahar) [inline likhithasai/assignment-1/Anagram.java]: It's nice to start with a javadoc explaining what the class does, its space and time complexity as it'd help me understand more your thought process.

File: likhithasai/assignment-1/Anagram.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+public class Anagram {
+
+    public boolean isAnagram(String str1, String str2) {
+		int n1 = str1.length();
+		int n2 = str2.length();
+        if (n1 != n2) {
+            //System.out.println("Not equal length");
```
Comment (schahar) [inline likhithasai/assignment-1/Anagram.java]: Nit : Remove leftover comments

File: likhithasai/assignment-1/TDDAnagram.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (schahar) [inline likhithasai/assignment-1/TDDAnagram.java]: Nit: Remove empty line

File: likhithasai/assignment-1/TDDAnagram.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import org.junit.Test;
+import org.junit.Assert.*;
```
Comment (schahar) [inline likhithasai/assignment-1/TDDAnagram.java]: No wildcard imports : https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports

File: likhithasai/assignment-1/TDDAnagram.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+import static org.junit.Assert.assertEquals;
```
Comment (schahar) [inline likhithasai/assignment-1/TDDAnagram.java]: Re-order static imports to appear before non-static imports : https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing

File: likhithasai/assignment-1/TDDAnagram.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class TDDAnagram {
```
Comment (schahar) [inline likhithasai/assignment-1/TDDAnagram.java]: Great to see TDD!

Comment (likhithasai) [reply likhithasai/assignment-1/TDDAnagram.java]: I added in some simple test cases for the most basic specification but I'm aware that there are edge case like empty string and one character string. I will be adding in more tests to cater to the optional challenges too.

File: likhithasai/assignment-1/TDDAnagram.java (hunk 5 above)
Comment (likhithasai) [reply likhithasai/assignment-1/TDDAnagram.java]: Oh yes, great point! I did learn about this isn detail when I did the Oracle Certification for Java SE8 programmer program but yes, in this particular case it was the result of the IDE fix. Will keep that in mind for future assignments
